data/sql_setup/prepare_schedule_a.sql

```diff
@@ -26,24 +26,31 @@ begin
         two_year_transaction_period_new = get_transaction_year(new.contb_receipt_dt, new.rpt_yr);
 
         if two_year_transaction_period_new >= start_year then
+            delete from ofec_sched_a_queue_new where sub_id = new.sub_id;
             insert into ofec_sched_a_queue_new values (new.*, timestamp, two_year_transaction_period_new);
         end if;
+
         return new;
     elsif tg_op = 'UPDATE' then
         two_year_transaction_period_new = get_transaction_year(new.contb_receipt_dt, new.rpt_yr);
         two_year_transaction_period_old = get_transaction_year(old.contb_receipt_dt, old.rpt_yr);
 
         if two_year_transaction_period_new >= start_year then
+            delete from ofec_sched_a_queue_new where sub_id = new.sub_id;
```

Comment:
I was thinking about this a bit more and am wondering if we're overthinking it or if I'm missing something still:  the `delete` statement as it is will remove all records that currently exist in the queue table(s) it is targeting that matches the `sub_id` of the record being inserted/updated/deleted in the table that caused the trigger to run.

Once these are in place there would only ever be 0 or 1 record(s) that could match the `sub_id` because anything existing in the queue table(s) would be removed by the `delete` statement before the new record to be processed would be inserted.

On the off chance that more than one somehow made it's way in though, we'd still want them all removed so we're only ever dealing with one record.  This is technically only an issue upon inserts into the `ofec_sched_*_*` tables due to the primary key constraint, but it's nice to know that this is the case across all operations from a consistency standpoint.

Furthermore, the `timestamp` doesn't exist on the record being handled by the trigger; that is calculated within the trigger itself and added to the copy of the record that is inserted into the queue tables.

I think I'm either seeing confusion about what's trying to be fixed or I'm still confused myself.  As I understand it, the issue is that multiple records with the same `sub_id` can sometimes exist in the schedule A queue tables we have, and if that's the case then when our nightly updates run that will cause unique key constraint errors with any `insert` operations into the `ofec_sched_a_*` tables. due to the primary key already existing in the table once the first one of those records with the same `sub_id` from the queue is processed.  As soon as the nightly update hits the second record to insert with the same `sub_id` that was processed before it will fail.  These `delete` statements in the trigger that helps manage the queue tables will prevent that from ever happening.

Does this help at all, or am I way out in left field? :-)